enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.enricher.fabric8;
+
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.ProbeBuilder;
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.enricher.api.MavenEnricherContext;
+import org.apache.commons.lang3.StringUtils;
+
+public class WebAppHealthCheckEnricher extends AbstractHealthCheckEnricher {
+
+
+    private static final int DEFAULT_DELAY_READNESS = 10;
+    private static final int DEFAULT_DELAY_LIVENESS = 180;
+
+
+    public WebAppHealthCheckEnricher(MavenEnricherContext buildContext) {
+        super(buildContext, "f8-healthcheck-webapp");
+    }
+
+    // Available configuration keys
+    private enum Config implements Configs.Key {
+
+        scheme {{
+            d = "HTTP";
+        }},
+        port {{
+            d = "8080";
+        }},
+        path {{
+            d = "";
+        }},
+        initialDelay {{
+            d = "-1";
+        }};
+
+        protected String d;
+
+        public String def() {
+            return d;
+        }
+    }
+
+    @Override
+    protected Probe getLivenessProbe() {
+        return getProbe(false);
+    }
+
+    @Override
+    protected Probe getReadinessProbe() {
+        return getProbe(true);
+    }
+
+    private boolean isApplicable() {
+        return getContext()
+            .hasPlugin("org.apache.maven.plugins", "maven-war-plugin") &&
+            StringUtils.isNotEmpty(Configs.asString(getConfig(Config.path)));
+    }
+
+    private Probe getProbe(boolean readiness) {
+        if (!isApplicable()) {
+            return null;
+        }
+
+        Integer port = getPort();
+        String scheme = getScheme().toUpperCase();
+        String path = getPath();
+
+        return new ProbeBuilder().
+            withNewHttpGet().withNewPort(port).withPath(path).withScheme(scheme).endHttpGet().
+            withInitialDelaySeconds(getInitialDelay(readiness)).build();
+
+    }
+
+    private int getInitialDelay(boolean readness) {
+        final int initialDelay = Configs.asInt(getConfig(Config.initialDelay));
+
+        if (initialDelay < 0) {
+            return readness ? DEFAULT_DELAY_READNESS : DEFAULT_DELAY_LIVENESS;
+        }
+
+        return  initialDelay;
```

Comment:
What if I wanted specify two different Initial Delay values for  Readiness and Liveness probes? Even the default values specified by you for Readiness and Liveness probes are different ( 100 and 1000). So, it should be allowed to specifiy two different Initial Delay values for the two probes.
For eg.,
```
<f8-healthcheck-webapp>
<config>
  <initialdelayReadiness>200</initialdelayReadiness>
  <initialDelayLiveness>400</initialDelayLiveness>
</config>
</f8-healthcheck-webapp>
```

Reply:
If we opt for this, then we can eliminate this getInitialDelay( ) method and change to
```
private Probe getProbe(INT InitialDelay) { ... }
```
and

Call it like
```
protected Probe getReadinessProbe() {
        int initialDelay = Configs.asInt(getConfig(Config.initialDelayReadiness));
        initialDelay = initialDelay < 0 ? DEFAULT_DELAY_READINESS : initialDelay;
        return getProbe(initialDelay);
}
```

Reply:
Correct me if I am wrong here.

Reply:
Yes, I agree with you, there is one enricher that is doing in the same way as I am doing here, but SpringBoot and thorntail are doing as you mention here, so I am changing.